### DISCUSSION TITLE
Differentiating between dialects?

### DISCUSSION THREAD

**Discussion by elken** [clojure-emacs/clojure-ts-mode#6] [Q&A] [answered]
Wasn't sure which category this would fall under, but I opted to phrase this as a direct question.

Now that the various modes are intended to be merged into a single `clojure-ts-mode`; what's going to be the preferred way to differentiate between clj, cljs, bb, nbb etc?

The obvious choice that springs out would be attempting to guess from the filename, but that seems like it could fail & obviously fails for cases where there is no file associated to the buffer (eg a scratch buffer or similar).

An example of where this would be most beneficial would be something like [lsp-tailwindcss](https://github.com/merrickluo/lsp-tailwindcss) that I'll only want in cljs but I don't want to connect to a pointless language server in every clojure file or bb script I open.

Thanks for all the hard work getting this together!

**Comment by dannyfreeman** [comment, marked as answer]
Hey sorry I missed this, I don't think I get notifications for discussion. I wasn't sure what the use case for different dialect modes were when there wasn't any distinct code in clojure-ts-mode for any of the modes. So I didn't include it initially. I understand the use case you present here though and have opened up an issue to track it: https://github.com/clojure-emacs/clojure-ts-mode/issues/12#issue-1730853750 Thank you for bringing it to my attention!

**Reply by elken** [reply]
No worries on both counts! I agree that I've not had much of a need for it until now, so I figured try and get a good solution in now :)

There is a button on the right for creating issues from discussions too
![image](https://github.com/clojure-emacs/clojure-ts-mode/assets/2872862/384e40e4-0a50-4bd2-bf02-c7d3ad7718ba)

**Reply by dannyfreeman** [reply]
Hey, I fixed this in commit 4dc853df16ba09d10dc3a648865e681679c17606

It should be available on melpa soon (not melpa stable until a new release is made)

**Reply by elken** [reply]
Oh sweet! I'll make a note to have a test of this soon :)
